RELEASE_PROCESS_OPENSEARCH.md

```diff
@@ -0,0 +1,322 @@
+
+### Release Manager
+
+The release managers to a specific OpenSearch release will be assigned through volunteer model. The request for release managers will be posted in [OpenSearch public slack workspace](https://opensearch.slack.com/archives/C0561HRK961) (under releases channel) and selected on first come first served (FCFS) model.
+Note: The release managers should be maintainers of a repo under OpenSearch GitHub organization.
+
+### Release issue
+
+This issue captures the state of the OpenSearch release, its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help. There are linked issues on components of the release where individual components can be tracked. More details are included in the Maintainers Release owner section.
+
+
+#### Release Issue Update
+
+The release issue is created by an [automation workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/releases.yml). Once the release manager is finalized, the release manager should be updating the created release issue. Sample [Release Issue 2.8.0](https://github.com/opensearch-project/opensearch-build/issues/3434)
+
+#### Component Issues and Campaigns
+
+Generate a release issue in each component repository that references the main release issue, and then update the components section of the release issue with the corresponding release issues created in each component repository. Sample [component issues](https://github.com/opensearch-project/opensearch-build/issues/3434#issue-1679791812) for a given release. Use the existing [component_release_template.md](https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/component_release_template.md) to create the component release issue.
+
+If exists any release specific issues/campaigns link it back to the release issue. Sample linked [issues/campaigns](https://github.com/opensearch-project/opensearch-build/issues/3434#issuecomment-1552138916)
+
+Note: This section is not required for a patch release.
+
+### Version Increment
+
+Versions are incremented as soon as development starts on a given version to avoid confusion. Following is the example that depicts the version increment.
+
+* OpenSearch: `main` = 3.0.0, `2.x` = 2.9.0, and `2.8` = 2.8.1
+* common-utils: `main` = 3.0.0.0, `2.x` = 2.9.0.0 and `2.8` = 2.8.1.0
+
+#### Version Increment Core
+
+To ensure the version incrementation process is handled correctly, it is important to follow up with the core team. Currently, the version incrementation is being done manually by an individual from the core repositories. Sample OpenSearch Version Increment [PR](https://github.com/opensearch-project/OpenSearch/pull/7864/).
```

Comment:
Cross reference a communication template here would be nice. How does a volunteer "follow up with the core team", is it a slack message with something simple like

```
[ANNOUNCE] Feature Freeze and new branch for OpenSearch 2.10

NOTICE: Branch 2.10 has been cut and versions updated to 2.11 on stable branch.

Please observe the normal rules:

* No new features may be committed to the branch.
* Only documentation patches, build patches and serious bug fixes may be
  committed to the branch. However, you should submit all patches you
  want to commit as pull requests first to give others the chance to review
  and possibly vote against them. Keep in mind that it is our
  main intention to keep the branch as stable as possible.
* All patches that are intended for the branch should first be committed
  to the unstable branch, merged into the stable branch, and then into
  the current release branch.
* Normal unstable and stable branch development may continue as usual.
  However, if you plan to commit a big change to the unstable branch
  while the branch feature freeze is in effect, think twice: can't the
  addition wait a couple more days? Merges of bug fixes into the branch
  may become more difficult.
* Only Github issues with Milestone v2.10
  and priority "Blocker" will delay a release candidate build.
```

Reply:
Just a quick thought, how about we add a section called as `### Slack Templates` and an RM will have a choice to use the templates or use his own vocabulary to coordinate in the channel ?
@bbarani @peterzhuamazon @nknize WDYT?

Reply:
Added `Communication Templates` section.
https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#communication-templates

We should improve this as we go.